### ISSUE TITLE
Poltergeist timeout

### ISSUE THREAD

**Issue by AlreadyTalk** [vitalie/webshot#6]
Good morning,
I have a issue with the webshot.
I'm running the project in my local machine.  I try to take a screenshot from my webpage, it gives my a time out from the poltergeist. If I try to take a the same page screenshot from the same page, but in some associate instance, doesn't give me the time out. 
I look for the answer in other forums and nothing.
Can someone help me please?
I'm using Poltergeist, Capybara and PhantomJS

**Comment by vgracia** [comment]
Yep same here. : (   It seems that timeout is hard to recover from.

**Comment by rgo** [comment]
It's not the solution for the timeouts(I think is a poltergeist/phantomJS problem) but it can helps to mitigate it.

@vitalie: You implemented a [timeout with sleep in the capture method](https://github.com/vitalie/webshot/blob/master/lib%2Fwebshot%2Fscreenshot.rb#L45). You can set it very high to avoid the timeouts but it's a pain to wait for working URLs

I was thinking in replace the sleep with this code:

``` ruby
def wait_until
  require "timeout"
  Timeout.timeout(Capybara.default_wait_time) do
    sleep(0.1) until value = yield
    value
  end
end
```

Code from: https://gist.github.com/jnicklas/d8da686061f0a59ffdf7

But instead of getting the value from a yield get it from the "visit url" or "page.driver.status_code"

What do you think?

**Comment by corehook** [comment]
Yep same here. : (

**Comment by mrdougwright** [comment]
Any fix yet?

**Comment by rgo** [comment]
@mrdougwright have you tried with latest version(0.1.0)?

It upgrades poltergeist and allows to requires phantomjs to 2.x.  If I'm not mistaken timeout problem was solved since this phantomjs version.

Regards

**Comment by mrdougwright** [comment]
Yes, and still getting timeout. I may open separate issue as I'm now seeing the webshots misplaced when called repeatedly from an each block.
